### PR TITLE
Replace deprecated request with node-fetch

### PR DESCRIPTION
**Pull request by sebastienbarbier**
Quick PR for maintenance, bumping 1.2.3 👍

- Replace deprecated request library with node-fetch
- Update minor dependencies
- Fix broken link within README

Ref : CPB-51857

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,21 +61,19 @@
   },
   "homepage": "https://github.com/sap/fsm-sdk#readme",
   "dependencies": {
-    "request": "^2.88.0",
-    "request-promise-native": "^1.0.5",
-    "uuid": "^3.3.2"
+    "@types/node-fetch": "^2.5.8",
```
Comment (GauSim) [inline package.json:64]: this should be a devDep no?

File: src/core-api.client.ts [hunk 2]
```diff
@@ -114,11 +115,19 @@ export class CoreAPIClient {
...
+  private _request<T>(uri: string, opt: RequestInit) {
     if (this._config.debug) {
-      console.log(`[httpRequest] outgoing options[${JSON.stringify(opt, null, 2)}]`);
+      console.log(`[httpRequest] outgoing ${uri} options[${JSON.stringify(opt, null, 2)}]`);
     }
-    return request(opt)
+    return fetch(uri, opt)
+      .then(response => {
+        const contentType = response.headers.get("content-type");
+        if (contentType && contentType.indexOf("application/json") !== -1) {
```
Comment (GauSim) [inline src/core-api.client.ts:125]:
```suggestion
        if (contentType && contentType.includes('application/json')) {
```

File: src/core-api.client.ts [hunk 3]
```diff
@@ -114,11 +115,19 @@ export class CoreAPIClient {
...
-  private _request<T>(opt: request.Options) {
+  private _request<T>(uri: string, opt: RequestInit) {
     if (this._config.debug) {
-      console.log(`[httpRequest] outgoing options[${JSON.stringify(opt, null, 2)}]`);
+      console.log(`[httpRequest] outgoing ${uri} options[${JSON.stringify(opt, null, 2)}]`);
     }
-    return request(opt)
+    return fetch(uri, opt)
+      .then(response => {
+        const contentType = response.headers.get("content-type");
```
Comment (GauSim) [inline src/core-api.client.ts:124]:
```suggestion
        const contentType = response.headers.get('content-type');
```

File: src/core-api.client.ts [hunk 4]
```diff
@@ -135,16 +144,20 @@ export class CoreAPIClient {
...
-        method !== 'DELETE' ? { dtos: RequestOptionsFacory.getDTOVersionsString([dtoName]) } : undefined,
-        additionalQs),
-      json: dtoData
-    });
+    const params = new URLSearchParams(Object.assign({},
+      RequestOptionsFacory._getRequestAccountQueryParams(token, this._config),
+      method !== 'DELETE' ? { dtos: RequestOptionsFacory.getDTOVersionsString([dtoName]) } : undefined,
+      additionalQs));
+
+    return await this._request<any>(
```
Comment (GauSim) [inline src/core-api.client.ts:152]: can we make any into a generic type to not lose type information ?

**Review by GauSim — CHANGES_REQUESTED** [review]

**Comment by GauSim** [comment]
👍🏽  cool PR

File: package.json (hunk 1 above)
Comment (sebastienbarbier) [reply package.json:64]: 🙈

**Comment by sebastienbarbier** [comment]
Thanks @GauSim for the review, great feedbacks as always, they are all applied and ready for an other look 👍

**Review by GauSim — APPROVED** [review]
🎉 🎉 🎉

**Pull request merged**